### ISSUE TITLE
Add Voikko for Finnish spell check

### ISSUE THREAD

**Issue by ilmaisin** [flathub/org.libreoffice.LibreOffice#122]
Could this flatpak be made to include the [Voikko](voikko/libreoffice-voikko) Finnish spell check tool?

**Comment by stbergmann** [comment]
So my understand is that, for all the various languages, the spell checking data used by LO is provided by hunspell from org.freedesktop.Platform//20.08.  While that runtime apparently contains hunspell data for lots of languages (`flatpak run --command=bash org.freedesktop.Platform//20.08 -c 'ls -d /usr/share/runtime/locale/*/share/*/hunspell | wc -l` reports 53), it does not contain any /usr/share/runtime/locale/fi/share/fi/hunspell.  I wonder what the general state of Finnish spell checking data is, is it lacking from hunspell in general (and not just in org.freedesktop.Platform//20.08)?  If yes, wouldn't it be better to add Finnish spell checking data to hunspell?  What do other flatpaks do that use the spell checking data from org.freedesktop.Platform?

**Comment by ilmaisin** [comment]
> So my understand is that, for all the various languages, the spell checking data used by LO is provided by hunspell from org.freedesktop.Platform//20.08. While that runtime apparently contains hunspell data for lots of languages (`flatpak run --command=bash org.freedesktop.Platform//20.08 -c 'ls -d /usr/share/runtime/locale/*/share/*/hunspell | wc -l` reports 53), it does not contain any /usr/share/runtime/locale/fi/share/fi/hunspell. I wonder what the general state of Finnish spell checking data is, is it lacking from hunspell in general (and not just in org.freedesktop.Platform//20.08)? If yes, wouldn't it be better to add Finnish spell checking data to hunspell? What do other flatpaks do that use the spell checking data from org.freedesktop.Platform?

Afaik voikko does not have anything to do with hunspell. Finnish needs little different spell-check algorithms.

**Comment by stbergmann** [comment]
The "stbergmann closed this in #158" auto-close was bogus.  But in a next step I'll provide an org.libreoffice.LibreOffice.BundledExtension.Voikko extension flatpak that will then close this issue.

**Comment by stbergmann** [comment]
> Any idea what other applications (flatpak'ed or not) do if they want Finnish spell checking?

I learned that other applications potentially use Enchant and its Voikko backend.  However, neither is that available in the org.freedesktop.Platform//20.08 runtime, nor could LibreOffice use it without modifications to the LibreOffice source code.

I have now filed <https://github.com/stbergmann/flathub/commit/e2aaf7193988e19304d5e036ef5db57f924c99e0> "Provide Voikko as a bundled extension for LibreOffice".

Of note is the following commit message excerpt:  "There appears to be no good reason that the official Finnish Spellchecker extension for LibreOffice, see <https://extensions.libreoffice.org/en/extensions/show/finnish-spell-checker-and-hyphenator-voikko> and <https://www.puimula.org/htp/ooo/voikko-win/5.0.2.20170827/voikko.oxt>, does not provide the necessary files to work on Linux in general, and thus in the LibreOffice flatpak in particular, see the mail thread starting at <https://lists.puimula.org/pipermail/libvoikko/2021-August/000955.html> '[libvoikko] Choice of native code inclusion in voikko.oxt for LibreOffice'.  But as long as that general issue is not addressed, provide it as a bundled flatpak extension, similar to how Linux distros provide it as a bundled extension in whatever distro package formats."
